src/components/Header.jsx

```diff
@@ -130,10 +141,35 @@ function Header() {
           <HeaderNavMenu links={links} />
         </Grid>
 
+        <Grid item>
+          <IconButton onClick={(e) => setAnchorEl(e.currentTarget)}>
+            <LanguageIcon />
+          </IconButton>
+        </Grid>
+
         <Grid item className={classes.authItem}>
-          <AuthZoomButton />
+          <AuthZoomButton t={t} />
         </Grid>
       </Grid>
+
+      <Menu
+        open={Boolean(anchorEl)}
+        anchorEl={anchorEl}
+        onClose={() => setAnchorEl(null)}
+        anchorOrigin={{ horizontal: 'center', vertical: 'bottom' }}
+        transformOrigin={{ horizontal: 'center', vertical: 'top' }}>
+        {Object.entries(languages).map(([k, v]) => (
```

Comment:
Nit: I suggest using slightly better names than `k` and `v` as it's not entirely clear what maps to what here. Maybe `lang` and `code`/`abbr`?